behind_ingensec.rb

```diff
@@ -0,0 +1,285 @@
+
+    return response
+  end
+
+  def do_check_bypass(fingerprint, host, ip, uri, proxies)
+
+    # Check for "misconfigured" web server on TCP/80.
+    if do_check_tcp_port(ip, 80, proxies)
+      response = do_simple_get_request_raw(ip, 80, false, host, uri, proxies)
+
+      if response != false
+        html = response.get_html_document
+        if fingerprint.to_s.eql? html.at('head').to_s
+          print_good("A direct-connect IP address was found: #{ip}")
+          return true
+        end
+      end
+    end
+
+    # Check for "misconfigured" web server on TCP/443.
+    if do_check_tcp_port(ip, 443, proxies)
+        response = do_simple_get_request_raw(ip, 443, true, host, uri, proxies)
+
+        if response != false
+          html = response.get_html_document
+          if fingerprint.to_s.eql? html.at('head').to_s
+            print_good("A direct-connect IP address was found: #{ip}")
+            return true
+          end
+        end
+      end
+
+    return false
+  end
+
+  def run
+    print_status('Passive gathering information...')
+
+    domain_name  = PublicSuffix.parse(datastore['HOSTNAME']).domain
+
+    # PrePost SEO
+    ip_records   = do_grab_domain_ip_history(domain_name, datastore['Proxies'])
+    ip_list      = ip_records if ip_records
+    print_status(" * PrePost SEO: #{ip_records.count.to_s} IP address found(s).")
+
+    # DNS Enum.
+    ip_records   = do_dns_enumeration(domain_name, datastore['THREADS'])
+    ip_list     |= ip_records if ip_records
+    print_status(" * DNS Enumeration: #{ip_records.count.to_s} IP address found(s).")
+    print_status()
+
+    # Cleaning the results.
+    print_status("Clean binarysec/ingensec server(s)...")
+    records      = []
+    ip_list.each do | ip |
+      a = do_dns_get_a(ip.to_s)
+      unless a.to_s.include? ("binarysec" || "ingensec")
+        unless ip.to_s.eql? "127.0.0.1"
+          records |= ip.to_a
+        end
+      end
+    end
+    print_good(" * TOTAL: #{records.count.to_s} IP address found(s) after cleaning.")
+    print_status()
+
+    # Processing bypass...
+    unless records.eql? false || records.empty?
+      print_status('Bypass BinarySec/IngenSec is in progress...')
+
+      # Initial HTTP request to the server (for <head> comparison).
+      print_status(' * Initial request to the original server for comparison')
+      response = do_simple_get_request_raw(
+        datastore['HOSTNAME'],
+        datastore['RPORT'],
+        datastore['SSL'],
+        nil,
+        datastore['URIPATH'],
+        datastore['PROXIES']
+      )
+
+      html     = response.get_html_document
+      head     = html.at('head')
+
+      ret_val  = false
+      records.each_with_index do | ip, index |
+        vprint_status(" * Trying: #{ip}")
+
+        ret_val = do_check_bypass(
+            head,
+            datastore['HOSTNAME'],
+            ip,
+            datastore['URIPATH'],
+            datastore['PROXIES']
+        )
+        break if ret_val.eql? true
```

Comment:
Record notes or vulns maybe instead of console-only?

Reply:
it's planned :)